src/Host.tsx

```diff
@@ -44,8 +44,36 @@ export const Host = ({ children, style }: IHostProps): JSX.Element => {
     }
   }, []);
 
-  const mount = (children: React.ReactNode): number => {
-    const key = Date.now();
+  //Keep track of all the keys we've generated to ensure uniqueness
+  let usedKeys: Array<string> = [];
+
+  //Makes a magical random key
+  const keyGenerator = (): string => {
+    return `portalize_${Math.random().toString(36).substr(2, 16)}-${Math.random()
+      .toString(36)
+      .substr(2, 16)}-${Math.random().toString(36).substr(2, 16)}`;
+  };
+
+  //Keeps making new keys until a unique one has been found
+  const getNewKey = (): string => {
+    let foundUniqueKey = false;
+    let newKey = '';
+    let tries = 0;
+    while (!foundUniqueKey && tries < 5) {
```

Comment:
What happen when we didn't find a unique key after 5 tries, we get an empty string?

Reply:
> What happen when we didn't find a unique key after 5 tries, we get an empty string?

Yeah.. totally open to suggestions here as to what our fallback key should be! The 'randomness' of the key generation probably means its super unlikely to have even 1 clash, let alone 5. But I agree, we should have some sort of fall back to something. Any ideas?

Reply:
Maybe 5 tries is too much, if it's super unlikely. Can we start with 3? 

Yes, let's fallback to Date.now() ahahah

Reply:
This is hilarious! haha ok no worries. Im sure we could 'work out' the probability of hitting 3 times, but I think its sooo unlikely